### ISSUE TITLE
[Feature Request/Bug] [Linux] Motion controls don't seem to be available when using DualSense

### ISSUE THREAD

**Issue by AMDBartek** [cemu-project/Cemu#221]
While using Cemu 4eaa600 motion controls don't seem to be available when using a DualSense controller (SDL API). It is not an issue with the controller as motion controls work in yuzu and other software.
![image](https://user-images.githubusercontent.com/46637209/189366375-7920bbc8-e727-40db-93cc-477dfa6c488d.png)

Also, the performance of native Linux Cemu seems to be rather low and has lower FPS than running older versions of Cemu in WINE. This is **not** a complaint but rather a question: Why does this occur (I would have thought it would have been the other way around)?
![image](https://user-images.githubusercontent.com/46637209/189367343-987ad00f-0eaa-4368-bb6b-79445cde5361.png)

Whilst using an older version of Cemu via Wine I would get around 70 FPS.

**Comment by AMDBartek** [comment]
Yep, no problem. I am using an Nvidia RTX 3060 Laptop GPU with Nvidia proprietary drivers (version 515) and my CPU is a MD Ryzen 5 5600H.

**Comment by AMDBartek** [comment]
I wasn't aware that there is an open source driver except Nouveau. How does the performance compare?

**Comment by Squall-Leonhart** [comment]
This is the case on Linux, ds4drv-cemuhook is required.

**Comment by AMDBartek** [comment]
Yes, but yuzu supports motion controls via SDL so it is possible. It would certainly be a nice feature to have.

**Comment by AMDBartek** [comment]
My apologies, I was unsure if this was a bug or if it was just unimplemented.

**Comment by AMDBartek** [comment]
It seems that this is a problem with the Cemu UI as setting the values manually in your controllers config XML file works.

```
<motion>true</motion>
<rumble>1</rumble>
```

After putting these under display_name motion controls work fine.

**Comment by AMDBartek** [comment]
Apologies, the performance is very good actually. I am just not the brightest and I didn't turn "FPS++" or whatever it's called back on. This is a perfectly satisfactory framerate for me with graphical enhancements.
![image](https://user-images.githubusercontent.com/46637209/190675803-656fcf11-b1b4-4232-b8cf-f069b2a4e901.png)

**Comment by AMDBartek** [comment]
After further testing, editing the config file does work and motion controls work fine but only over USB, motion controls do not work over Bluetooth in Linux but they did on Windows. Strange

**Comment by Crementif** [comment]
Ah, good to know about the SDL working better over wireless controls. And glad you found the fix for your performance issue.

**Comment by AMDBartek** [comment]
I think it might be something to do with SDL, when removing it from vcpkg.json and using the libsdl2-dev pacakge from Ubuntu repository motion controls work over wireless as well as wired (https://github.com/cemu-project/Cemu/issues/107#issuecomment-1252648524).

**Comment by Amnesia1000** [comment]
@AMDBartek could you tell me how did you make the DualSense gyroscope work in cemu?

**Comment by Masamune3210** [comment]
same way you make the DS4's gyro work in Cemu, IIRC. Enable the UDP server in DS4Windows and tell cemu to grab the motion data from it

**Comment by Squall-Leonhart** [comment]
you don't use ds4windows.

**Comment by Masamune3210** [comment]
re-read the comment, hence deleting my comment lol

**Comment by Amnesia1000** [comment]
@Squall-Leonhart I have installed ds4drv-cemuhook but it still does not recognize the gyroscope.
I have also installed evdevhook, and when I run it, the DualSense appears as a motion control, but Cemu still does not recognize it.

**Comment by jcharaoui** [comment]
> I think it might be something to do with SDL, when removing it from vcpkg.json and using the libsdl2-dev pacakge from Ubuntu repository motion controls work over wireless as well as wired ([#107 (comment)](https://github.com/cemu-project/Cemu/issues/107#issuecomment-1252648524)).

I was able to compile Cemu (git `HEAD`) with this change (replacing `sdl2` from vcpkg.json with libsdl2-dev) on Debian testing, (using clang-14), but unfortunately that didn't make motion controls available with my Steam controller (both wireless and wired). The XML editing trick didn't help either.

**Comment by Adesin-fr** [comment]
Same for me...
I can't understand what's happening : 
A few days ago I was on ubuntu 22.04 : I built Cemu from HEAD version, removing sdl2 from vcpkg, and it worked. Then I had the silly idea of upgrading ubuntu to 22.10.... But I had to revert to 22.04, so I did reinstall my ubuntu from scratch, keeping my user folder. 
I then rebuilt Cemu with the same folder (so sdl2 is still removed from vcpkg), but now motion doesn't work anymore :(

**Comment by Squall-Leonhart** [comment]
you need sdl, period,  this will become simpler when 2.26 is pulled in and hidapi is turned on where relevant.

**Comment by Adesin-fr** [comment]
libudev was already installed (and libsdl2-dev also...).
Just installed hidapi ( sudo apt install libhidapi-dev ), and rebuild completely (deleted build folder contents), but still no luck

**Comment by Adesin-fr** [comment]
Managed to get it working ! 
I had to install libsdl2 system package,  then launch the latest AppImage with this command : 
`LD_PRELOAD=/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0 ./Cemu-2.0-25-x86_64.AppImage`

Be sure to check that the path of the file (`/usr/lib/x86_64-linux-gnu/libSDL2-2.0.so.0`) in your system . If not sure, search it with `find / -name libSDL2-2.0.so.0`

**Comment by MicahBird** [comment]
I was having troubles with getting motion controls working with a DS4 on Ubuntu 22.10, and tried all of the previous tips/tricks to no avail... _However_ I found that it worked after manually building the AppImage from the latest main branch (using Github Actions)! I also had to manually remove and re-add the controller in the Cemu ui, but then the motion controls checkbox was no longer grayed out and the motion controls work flawlessly!
